### DISCUSSION TITLE
Tox and Deepsource

### DISCUSSION THREAD

**Discussion by Yobmod** [gitpython-developers/GitPython#1293] [General]
Hi,

Are any contributors still using Tox to test locally?

I'm using pytest plugins that run flake8 and mypy all-in-one, or I run them seperately (if a test fails i don't care about flake8 until its fixed). 
Locally I just test against whichever python I am using, as pushing to github actions tests in parallel, which is faster than running tox for me.

So with it not used in CI anymore, we could remove tox as a test-dependancy, unless people find it useful.
(I don't know if #1090 is still an issue with tox/make, as I don't use docker.)

Also is anyone still using Deepsource? (by anyone I guess I mean @Byron, lol) Or is it safe to remove the skipcq comments for that?

**Comment by Byron** [comment]
Thanks for bringing this up. I have the feeling that `pytest` is the norm by now and other tooling isn't anything that we have to support anymore as they don't provide a better development experience. Hence I am absolutely happy to remove `tox` from the project and the documentation - the README could probably use an update, maybe the contributing guide as well.

> Also is anyone still using Deepsource? 

I think it has been deactivated a while ago due to the noise it generated, please feel free to remove remaining traces of it :).
